### PR TITLE
DENG-179 Add columns to mobile_search_clients_daily_v1

### PR DESCRIPTION
**Pull request by lucia-vargas-a**
This PR adds 2 columns (install_source and adjust_network) to `search_derived.mobile_search_clients_daily_v1`, to allow for the segmentation of Firefox Android new profiles.

It will be followed by a schema deploy and a backfill from 2021-01-01 until the current date.

For more context, this table is source for search metrics for unified_metrics and collects data for all browsers and channels, therefore is the best source for this implementation, instead of creating another derived dataset.

After running health checks on a subset for July 2022 backfilled in **analysis.mobile_search_clients_daily_v2**, I found a few differences versus  the production dataset in the count of clients:

Fenix reduced 0.2%
Fennec| reduced 0.7%
FirefoxForFireTV reduce 0.8%
Klar increased 30%
Focus| increased 31%

FirefoxConnect, Zerda_cn, FirefoxReality_oculusvr, Zerda, Other, WebXR, Firefox, Scryer, Lockbox, FirefoxReality_wavevr, FirefoxReality_googlevr have exact dsame counts.

Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [x] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

**Comment by lucia-vargas-a** [comment]
> I'd recommend actually running the generator to update the `query.sql` here since it hasn't been removed yet. It's probably more confusing to have a generated query that doesn't match the generator. Removing it is also an option (though I'd recommend doing that in a separate PR)

Thanks @ANich, I updated the query and the tests.
But tests are failing, I think it might be because the schema is not deployed yet... Do you know if I should deploy the schema at this point? Or I maybe i am missing something in the tests?
`metrics_adjust_network does not exist in STRUCT<search_default_engine_code STRING, search_default_engine_submission_url STRING>`

Btw, shall we update the docs for a guidance on how to update a dataset & tests via the sql generators? I didn't find much documentation on it, happy to do that.

**Comment by ANich** [comment]
> > I'd recommend actually running the generator to update the `query.sql` here since it hasn't been removed yet. It's probably more confusing to have a generated query that doesn't match the generator. Removing it is also an option (though I'd recommend doing that in a separate PR)
> 
> Thanks @ANich, I updated the query and the tests. But tests are failing, I think it might be because the schema is not deployed yet... Do you know if I should deploy the schema at this point? Or I maybe i am missing something in the tests? `metrics_adjust_network does not exist in STRUCT<search_default_engine_code STRING, search_default_engine_submission_url STRING>`
> 
> Btw, shall we update the docs for a guidance on how to update a dataset & tests via the sql generators? I didn't find much documentation on it, happy to do that.

Taking a look at CI, it seems like this is the error:
```
In Expected, not in Result - .0.install_source
In Expected, not in Result - .0.adjust_network
In Expected, not in Result - .1.install_source
In Expected, not in Result - .1.adjust_network
In Expected, not in Result - .2.install_source
In Expected, not in Result - .2.adjust_network
...
```

I think you'll have to change the output CTE in `mobile_search_clients_daily.template.sql` to include the new fields

**Comment by lucia-vargas-a** [comment]
> I'd recommend actually running the generator to update the `query.sql` here since it hasn't been removed yet. It's probably more confusing to have a generated query that doesn't match the generator. Removing it is also an option (though I'd recommend doing that in a separate PR)

@ANich  for info, does this mean that there is plan to remove this file and rely on generator running in the CI?

**Comment by ANich** [comment]
> @ANich for info, does this mean that there is plan to remove this file and rely on generator running in the CI?

Yeah, I can open a follow-up PR to do that

**Comment by lucia-vargas-a** [comment]
@ANich PR ready for a second look please,
I added some context to the description.

**Comment by lucia-vargas-a** [comment]
I'll need to deploy schema for the CI

File: sql_generators/search/templates/fenix_metrics.template.sql [hunk 1]
```diff
@@ -19,6 +19,33 @@ metrics_{{ namespace }} AS (
     '{{ channel }}' AS channel,
     normalized_os AS os,
     client_info.android_sdk_version AS os_version,
+    CASE
+    WHEN
+      ARRAY_AGG(metrics.string.metrics_adjust_network)[SAFE_OFFSET(0)] NOT IN (
```
Comment (ANich) [inline sql_generators/search/templates/fenix_metrics.template.sql:24]: Current test failure is related to this part, this is an aggregation so you'll have to add a group-by and potentially other aggregations here

**Comment by lucia-vargas-a** [comment]
To be implemented via [Proposal for First Attribution](https://docs.google.com/document/d/12bj4DhCybelqHVgOVq8KJlzgtbbUw3f68palNrv-gaM/edit#heading=h.7ot11lanr2k9)

**Pull request closed without merging**